pkg/zpdb/pod_observer_test.go

```diff
@@ -0,0 +1,114 @@
+package zpdb
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	k8sfake "k8s.io/client-go/kubernetes/fake"
+)
+
+func newPodObserverTestCase() (*k8sfake.Clientset, *PodObserver) {
+	client := k8sfake.NewClientset()
+	observer := NewPodObserver(client, testNamespace, log.NewNopLogger())
+	return client, observer
+}
+
+func createTestPod(name, namespace string) *corev1.Pod {
+	return &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			UID:       types.UID(fmt.Sprintf("uid-%s", name)),
+		},
+	}
+}
+
+// TestObserver_NewPdbObserver- basic constructor and life cycle test
+func TestObserver_NewPodObserver(t *testing.T) {
+	_, observer := newPodObserverTestCase()
+
+	require.NotNil(t, observer)
+	require.NotNil(t, observer.PodEvictCache)
+	require.NotNil(t, observer.podsFactory)
+	require.NotNil(t, observer.podsInformer)
+	require.NotNil(t, observer.logger)
+	require.NotNil(t, observer.stopCh)
+	require.NoError(t, observer.Start())
+
+	// Ensure that stopCh has been opened
+	select {
+	case <-observer.stopCh:
+		t.Fatal("stopCh should not be closed initially")
+	default:
+		// Expected - channel is open
+	}
+
+	observer.Stop()
+
+	// Ensure that stopCh has been closed
+	select {
+	case <-observer.stopCh:
+		// Expected - channel is closed
+	case <-time.After(100 * time.Millisecond):
+		t.Fatal("stopCh should be closed after Stop()")
+	}
+}
+
+// TestObserver_PodEvents validates the pod eviction cache is invalidated on pod changes
+func TestObserver_PodEvents(t *testing.T) {
+	client, observer := newPodObserverTestCase()
+	require.NoError(t, observer.Start())
+	defer observer.Stop()
+
+	pod := createTestPod("test-pod", testNamespace)
+
+	// Add pod to fake client - this should trigger the informer and invalidate the cache
+	observer.PodEvictCache.RecordEviction(pod)
+	require.True(t, observer.PodEvictCache.HasPendingEviction(pod))
+	_, err := client.CoreV1().Pods(testNamespace).Create(context.Background(), pod, metav1.CreateOptions{})
```

Comment:
This took me a second to understand - might be worth adding a comment explaining that the newly created pod won't be in a ready state, so this will trigger us expiring it from the cache